src/components/algolia-search/index.tsx

```diff
@@ -0,0 +1,195 @@
+
+  React.useEffect(() => {
+    if (!formRef.current || !panelRef.current || !inputRef.current) {
+      return undefined
+    }
+
+    const { onTouchStart, onTouchMove } = getEnvironmentProps({
+      formElement: formRef.current,
+      inputElement: inputRef.current,
+      panelElement: panelRef.current,
+    })
+
+    window.addEventListener('touchstart', onTouchStart)
+    window.addEventListener('touchmove', onTouchMove)
+
+    return () => {
+      window.removeEventListener('touchstart', onTouchStart)
+      window.removeEventListener('touchmove', onTouchMove)
+    }
+  }, [getEnvironmentProps, formRef, inputRef, panelRef])
+
+  /**
+   * Allow focusing of the search input on pressing '/'
+   */
+  useFocusOnKeyClick(inputRef, '/')
+
+  return (
+    <div className={s.root} {...autocomplete.getRootProps({})}>
+      <form
+        ref={formRef}
+        className={s.form}
+        {...autocomplete.getFormProps({ inputElement: inputRef.current })}
+      >
+        <div className={s.inputWrapperPrefix}>
+          <label {...autocomplete.getLabelProps({})}>
+            <button className={s.submitButton} type="submit" title="Submit">
+              <IconSearch16 />
+            </button>
+          </label>
+        </div>
+        <input
+          className={s.input}
+          ref={inputRef}
+          {...autocomplete.getInputProps({ inputElement: inputRef.current })}
+        />
+        <div className={s.inputWrapperSuffix}>
+          <button
+            className={s.clearButton}
+            title="Clear"
+            type="reset"
+            hidden={!autocompleteState.query}
+          >
+            <IconX16 />
+          </button>
+        </div>
+        {autocompleteState.query ? null : (
+          <IconSlashSquare16
+            className={s.keyboardHintIcon}
+            title="Type '/' to Search"
+          />
+        )}
+      </form>
+
+      {autocompleteState.isOpen && (
+        <div
+          ref={panelRef}
+          className={s.panel}
+          {...autocomplete.getPanelProps({})}
+        >
+          <div className={s.panelLayout}>
+            {autocompleteState.collections.map((collection, index) => {
+              const { source, items } = collection
+
+              return (
+                <section key={`source-${index}`}>
+                  {items.length > 0 && (
+                    <ul className={s.list} {...autocomplete.getListProps()}>
+                      {items.map((item) => {
+                        return (
+                          <li
+                            key={item.objectID}
+                            className={s.item}
+                            {...autocomplete.getItemProps({ item, source })}
+                          >
+                            <HitWrapper
+                              hit={item}
+                              className={s.itemLink}
+                              onHitClick={() => {
+                                autocomplete.setQuery('')
```

Comment:
When you select a result from the hit box, is the search input supposed to clear? Right now I'm seeing that the previous input persists even after navigating to the new page.

Reply:
Ooh good catch. It looks like when you navigate via the arrow keys / enter it's not clearing, but navigating via mouse is clearing. Will dig in...